gtdynamics/factors/TorqueFactor.h

```diff
@@ -42,11 +42,11 @@ namespace gtdynamics {
  *
  * @param joint JointConstSharedPtr to the joint
  */
-inline gtsam::ExpressionFactor<double> TorqueFactor(
+inline gtsam::NoiseModelFactor::shared_ptr TorqueFactor(
     const gtsam::noiseModel::Base::shared_ptr &cost_model,
     const JointConstSharedPtr &joint, size_t k = 0) {
-  return gtsam::ExpressionFactor<double>(cost_model, 0.0,
-                                         joint->torqueConstraint(k));
+  return boost::make_shared<gtsam::ExpressionFactor<double>>(
+      cost_model, 0.0, joint->torqueConstraint(k));
```

Comment:
Hardcoding? That's usually a big no. Any reason this isn't just an optional function parameter?

Reply:
I don't get it. Is it the 0.0 that you are referring to? The torqueConstraint returns an expression, and the constraint itself basically forces the expression to be 0.

Reply:
Gotcha. Then that should be a comment `The expression constrains the forces to be 0`.

Reply:
It's a misunderstanding. The `torqueConstraint` and all other such functions returns an expression g(x). And the EqualityConstraint to create is basically g(x)=0.